packages/core/src/browser/tree/tree-view-welcome-widget.tsx

```diff
@@ -0,0 +1,245 @@
+/********************************************************************************
+ * Copyright (c) 2020 SAP SE or an SAP affiliate company and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+import React = require('react');
+import { inject, injectable } from 'inversify';
+import { CommandRegistry } from '../../common';
+import { ContextKeyService } from '../context-key-service';
+import { TreeModel } from './tree-model';
+import { TreeWidget } from './tree-widget';
+
+interface ViewWelcome {
+    view: string;
+    content: string;
+    when?: string;
+    order: number;
+}
+
+interface IItem {
+    readonly welcomeInfo: ViewWelcome;
+    visible: boolean;
+}
+
+interface ILink {
+    readonly label: string;
+    readonly href: string;
+    readonly title?: string;
+}
+
+type LinkedTextItem = string | ILink;
+
+@injectable()
+export class TreeViewWelcomeWidget extends TreeWidget {
+
+    @inject(CommandRegistry)
+    protected readonly commands: CommandRegistry;
+
+    @inject(ContextKeyService)
+    protected readonly contextService: ContextKeyService;
+
+    private viewWelcomeNodes: React.ReactNode[] = [];
+    private defaultItem: IItem | undefined;
+    private items: IItem[] = [];
```

Comment:
Generally, it's better to have `protected` everywhere instead of `private`. Just think about it, what if you would like to quickly customize something on your side, `protected` visibility could come in handy. Please change. Thanks!